activerecord/lib/active_record/relation/predicate_builder/range_handler.rb

```diff
@@ -14,17 +14,25 @@ def initialize(predicate_builder)
       end
 
       def call(attribute, value)
-        begin_bind = predicate_builder.build_bind_attribute(attribute.name, value.begin)
-        end_bind = predicate_builder.build_bind_attribute(attribute.name, value.end)
-        if value.begin.respond_to?(:infinite?) && value.begin.infinite?
-          if value.end.respond_to?(:infinite?) && value.end.infinite?
-            attribute.not_in([])
+        if !infinity?(value.begin) && !infinity?(value.end) && value.begin > value.end
+          return attribute.in([])
+        end
+
+        begin_bind = predicate_builder.build_boundable_attribute(attribute.name, value.begin)
+        end_bind = predicate_builder.build_boundable_attribute(attribute.name, value.end)
+        if infinity?(value.begin) || begin_bind.nil?
+          if infinity?(value.end) || end_bind.nil?
+            if begin_bind.nil? && value.begin > 0 || end_bind.nil? && value.end < 0
```

Comment:
I've removed converting out-of-range values to infinity.
So this line will check out-of-range values only, doesn't affect open-ended ranges.